### DISCUSSION TITLE
Question about builder passing null will send unnecessary info over the network?

### DISCUSSION THREAD

**Discussion by shiywang** [aws/aws-sdk-java-v2#3859] [General]
So we have  APIRequest has a, b, c, but c could be potential `null`  we are thinking about write like this:

```
APIRequest.Builder builder = APIRequest.builder()
                .fieldA(A)
                .fieldB(B);
 
if(APIRequest. getC() != null) {
    builder.fieldC(APIRequest.getC());
}
return builder.build();
```

instead of 
```
return APIRequest.builder()
                .fieldA(A)
                .fieldB(B)
                .fieldC(APIRequest.getC()).build();
```
Are those two the same? which one is better? (option 2 looks def cleaner) but if C is null, will the second option send more data via network? or it depends on the implementation of client?

**Comment by debora-ito** [comment]
Passing a null value forward can lead to NullPointerExceptions, which can be hard to troubleshoot. Having null checks is a best practice and there are several ways to do them, but implementation details are not in the scope of this channel. I recommend you search for Java programming best practices regarding nulls.
